x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -34,76 +38,88 @@ function updateEnvironmentUrl(
   });
 }
 
-const SEPARATOR_OPTION = {
-  text: `- ${i18n.translate(
-    'xpack.apm.filter.environment.selectEnvironmentLabel',
-    { defaultMessage: 'Select environment' }
-  )} -`,
-  disabled: true,
-};
-
-function getOptions(environments: string[]) {
-  const environmentOptions = environments
-    .filter((env) => env !== ENVIRONMENT_NOT_DEFINED.value)
-    .map((environment) => ({
-      value: environment,
-      text: environment,
-    }));
-
-  return [
-    ENVIRONMENT_ALL,
-    ...(environments.includes(ENVIRONMENT_NOT_DEFINED.value)
-      ? [ENVIRONMENT_NOT_DEFINED]
-      : []),
-    ...(environmentOptions.length > 0 ? [SEPARATOR_OPTION] : []),
-    ...environmentOptions,
-  ];
```

Comment:
Can the user still select "Not defined" and "All" environments?

Reply:
@sqren considering this https://github.com/elastic/kibana/issues/71239 do you have any concern about removing the "Not defined" option from the dropdown?

Reply:
we have 2 options:

1. Hard code the "Not defined" option and display it always 
2. Create a new search to determine if there is a `service` without a `service.environment`  and based on the results add the "Not defined" option. 
For example, 
 ```
{
  "size": 0,
  "terminate_after": 1,
  "query": {
    "bool": {
      "must_not": [
        {"exists": {"field": "service.environment"}}
      ]
    }
  }
}
```
With option 2 we mimic the experience we provide today, the only downside is one more request but it should be fast

Reply:
I think it's better if you can do option 2. It should be fast enough.
But won't it cause problems for your suggestion component that now have to be concerned with a secondary request?